### PR TITLE
Restore CI testing, fix tests

### PR DESCRIPTION
**Pull request by fregante**
Tests are failing though, which is strange because this works correctly locally: `dot-json a.json b c`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  test:
+    name: Node.js ${{ matrix.node-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 20
+          - 18
+          - 16
```
Comment (fregante) [inline .github/workflows/ci.yml:14]: - 20: current
- 18: lts
- 12: the version that was supposedly compatible when this package was being developed

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  test:
+    name: Node.js ${{ matrix.node-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 20
+          - 18
+          - 12
```
Comment (fregante) [inline .github/workflows/ci.yml:14]: - 20: current
- 18: lts
- 12: the version that was supposedly compatible when this package was being developed, can change it

File: lib/dot-json.js [hunk 3]
```diff
@@ -56,7 +56,7 @@ module.exports = function(file) {
 				_this._indent = detectIndent(_this.content()).indent;
 			}
 			catch (e) {
-				_this._indent = 2
+				_this._indent = '  '
```
Comment (fregante) [inline lib/dot-json.js]: The tests were alternating failure, probably depending on whether the file existed or not.

This function now always returns a string, never a number.

CI passes now: https://github.com/fregante/dot-json/commits/ci

**Comment by maikelvl** [comment]
@fregante Nice work! 💯

**Pull request merged**